rapid7/metasploit-framework
# Add module for Rails "DoubleTap" vulnerability

This PR adds an auxiliary module that exploits a path traversal vulnerability in Ruby on Rails `PathResolver` (versions <= 5.2.2). The impact is limited to calls to render which render file contents without a specified accept format.

## Verification

  1. Start a Rails server using a vulnerable version 
  2. Start msfconsole
  3. Do: ```use auxiliary/gather/rails_doubletap_file_read```
  4. Do: ```set ROUTE /your_route```
  5. Do: ```set RHOSTS target```
  6. Do: ```set TARGET_FILE /absolute/path/to/remote/file.txt```
  7. Do: ```run```
  8. If everything goes smoothly, you should get the contents of the remote file printed to the console.

## Light reading material
- https://mobile.twitter.com/iblueconnection/status/1107702203349979136
- https://github.com/mpgn/Rails-doubletap-RCE
- https://hackerone.com/reports/473888
- https://groups.google.com/forum/#!topic/rubyonrails-security/pFRKI96Sm8Q
